streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -5129,18 +5138,18 @@ object ZStream extends ZStreamPlatformSpecificConstructors {
    * Creates a stream by peeling off the "layers" of a value of type `S`
    */
   def unfold[S, A](s: => S)(f: S => Option[(A, S)])(implicit trace: Trace): ZStream[Any, Nothing, A] =
-    unfoldChunk(s)(f(_).map { case (a, s) => Chunk.single(a) -> s })
+    unfoldChunk(s)(f(_).map { case (a, s0) => Chunk.single(a) -> s0 })
```

Comment:
While I agree with the semantic changes (here and below), I think it's better to revert them just to keep git history as "clean" as possible.

I had to debug ZStream issues in the past and it was a bit time consuming having to double-check whether changes like these didn't accidentally introduce a bug